trieste/acquisition/rule.py

```diff
@@ -1103,6 +1115,380 @@ def state_func(
         return state_func
 
 
+class UpdateableSearchSpace(SearchSpace):
+    """A search space that can be updated."""
+
+    @abstractmethod
+    def reinitialise(
+        self,
+        models: Optional[Mapping[Tag, ProbabilisticModelType]] = None,
+        datasets: Optional[Mapping[Tag, Dataset]] = None,
+    ) -> None:
+        """Reinitialise the search space using the given models and datasets."""
+        ...
+
+    @abstractmethod
+    def update(
+        self,
+        models: Optional[Mapping[Tag, ProbabilisticModelType]] = None,
+        datasets: Optional[Mapping[Tag, Dataset]] = None,
+    ) -> None:
+        """Update the search space using the given models and datasets."""
+        ...
+
+    def get_single_model_and_dataset(
+        self,
+        models: Optional[Mapping[Tag, ProbabilisticModelType]] = None,
+        datasets: Optional[Mapping[Tag, Dataset]] = None,
+    ) -> Tuple[Optional[ProbabilisticModelType], Optional[Dataset]]:
+        """Get a single model and dataset from the given arguments."""
+        if models is None:
+            model = None
+        elif OBJECTIVE in models.keys():
+            model = models[OBJECTIVE]
+        else:
+            raise ValueError(f"""if models is provided, it must contain the key {OBJECTIVE}""")
+
+        if datasets is None:
+            dataset = None
+        elif OBJECTIVE in datasets.keys():
+            dataset = datasets[OBJECTIVE]
+        else:
+            raise ValueError(f"""if datasets is provided, it must contain the key {OBJECTIVE}""")
+
+        return model, dataset
+
+
+UpdateableSearchSpaceType = TypeVar("UpdateableSearchSpaceType", bound=UpdateableSearchSpace)
+""" A type variable bound to :class:`UpdateableSearchSpace`. """
+
+
+class MultiTrustRegion(
+    AcquisitionRule[
+        types.State[Optional["MultiTrustRegion.State"], TensorType],
+        SearchSpace,
+        ProbabilisticModelType,
+    ],
+    Generic[ProbabilisticModelType, UpdateableSearchSpaceType],
+):
+    """Abstract class for multi trust region acquisition rules."""
```

Comment:
a bit more explanation would be useful, e.g. its a batch algorithm where each element in the batch is optimised with its own TR, with optimisation done in parallel

Reply:
Thanks. I still need to check and update the docstrings.